fleet_vehicle_inspection/models/fleet_vehicle_inspection.py

```diff
@@ -0,0 +1,196 @@
+        index=True,
+        readonly=True,
+        track_visibility="onchange",
+        default="draft",
+        help=" * Draft: not confirmed yet.\n"
+        " * Confirmed: inspection has been confirmed.\n"
+        " * Cancelled: has been cancelled, can't be confirmed anymore.",
+    )
+
+    vehicle_id = fields.Many2one(
+        "fleet.vehicle",
+        "Vehicle",
+        help="Fleet Vehicle",
+        required=True,
+        states=READONLY_STATES,
+    )
+
+    odometer_id = fields.Many2one(
+        "fleet.vehicle.odometer",
+        "Odometer",
+        help="Odometer measure of the vehicle at the moment of this log",
+    )
+
+    odometer = fields.Float(
+        compute="_get_odometer",
+        inverse="_set_odometer",
+        string="Last Odometer",
+        help="Odometer measure of the vehicle at the moment of this log",
+        stored=True,
+        states=READONLY_STATES,
+    )
+
+    odometer_unit = fields.Selection(
+        [("kilometers", "Kilometers"), ("miles", "Miles")],
+        "Odometer Unit",
+        default="kilometers",
+        required=True,
+        states=READONLY_STATES,
+    )
+
+    date_inspected = fields.Datetime(
+        "Inspection Date",
+        required=True,
+        default=fields.Datetime.now,
+        help="Date when the vehicle has been inspected",
+        copy=False,
+        states=READONLY_STATES,
+    )
+
+    inspected_by = fields.Many2one(
+        "res.partner",
+        "Inspected By",
+        track_visibility="onchange",
+        states=READONLY_STATES,
+    )
+
+    direction = fields.Selection(
+        selection=[("in", "IN"), ("out", "OUT")], default="out", states=READONLY_STATES,
+    )
+
+    note = fields.Html("Notes", states=READONLY_STATES)
+
+    inspection_line_ids = fields.One2many(
+        "fleet.vehicle.inspection.line",
+        "inspection_id",
+        string="Inspection Lines",
+        copy=True,
+        auto_join=True,
+        states=READONLY_STATES,
+    )
+
+    result = fields.Selection(
+        [("todo", "Todo"), ("success", "Success"), ("failure", "Failure")],
+        "Inspection Result",
+        default="todo",
+        compute="_compute_inspection_result",
+        readonly=True,
+        copy=False,
+        store=True,
+    )
+
+    @api.depends("inspection_line_ids", "state")
+    def _compute_inspection_result(self):
+        for inspection in self:
+            if inspection.inspection_line_ids:
+                if inspection.inspection_line_ids.filtered(
+                    lambda x: x.result == "todo"
+                ):
+                    inspection.result = "todo"
+                    continue
+                elif inspection.inspection_line_ids.filtered(
+                    lambda x: x.result == "failure"
+                ):
+                    inspection.result = "failure"
+                else:
+                    inspection.result = "success"
+            else:
+                inspection.result = "todo"
```

Comment:
I'm nitpicking here, but I would take the opportunity to rewrite this method like this:

```suggestion
    def _compute_inspection_result(self):
        for rec in self:
            if rec.inspection_line_ids:
                if any(l.result == "todo" for l in rec.inspection_line_ids):
                    rec.result = "todo"
                elif any(l.result == "failure" for l in rec.inspection_line_ids):
                    rec.result = "failure"
                else:
                    rec.result = "success"
            else:
                rec.result = "todo"
```

- `rec` is only 3 chars, and used across many computed methods regardless of the model
- `any` is faster than scanning the whole recordset with `filtered`